nginx.sh

```diff
@@ -0,0 +1,75 @@
+if [ -f "huly.conf" ]; then
+    source "huly.conf"
+fi
+
+# Check for --recreate flag
+RECREATE=false
+if [ "$1" == "--recreate" ]; then
+    RECREATE=true
+fi
+
+# Handle nginx.conf recreation or updating
+if [ "$RECREATE" == true ]; then
+    cp .template.nginx.conf nginx.conf
+    echo "nginx.conf has been recreated from the template."
+else
+    if [ ! -f "nginx.conf" ]; then
+        echo "nginx.conf not found, creating from template."
+        cp .template.nginx.conf nginx.conf
+    else
+        echo "nginx.conf already exists. Only updating server_name, listen and proxy_pass."
+        echo "Run with --recreate to fully overwrite nginx.conf."
+    fi
+fi
+
+# Update server_name and proxy_pass
+sed -i "s|server_name .*;|server_name ${HOST_ADDRESS};|" ./nginx.conf
+sed -i "s|proxy_pass .*;|proxy_pass http://${HTTP_BIND:-127.0.0.1}:${HTTP_PORT};|" ./nginx.conf
+
+# Update listen directive to either port 80 or 443, while preserving IP address
```

Comment:
Some of commands below fail with errors, here is errors log:
```
 Generating nginx.conf...
nginx.conf not found, creating from template.
sed: 1: "./nginx.conf": invalid command code .
sed: 1: "./nginx.conf": invalid command code .
sed: 1: "s|(listen )(.*:)?[0-9]+ ...": \1 not defined in the RE
sed: 1: "s|(listen )(.*:)?([0-9] ...": \1 not defined in the RE
grep: invalid option -- P
usage: grep [-abcdDEFGHhIiJLlMmnOopqRSsUVvwXxZz] [-A num] [-B num] [-C[num]]
	[-e pattern] [-f file] [--binary-files=value] [--color=when]
	[--context[=num]] [--directories=action] [--label] [--line-buffered]
	[--null] [pattern] [file ...]
Enabling SSL; removing HTTP to HTTPS redirect block...
Do you want to run 'nginx -s reload' now to load your updated Huly config? (Y/n): n
You can run 'nginx -s reload' later to load your updated Huly config.
```

Reply:
Does my fix work?

Reply:
sed errors are gone, but grep error is still there

Reply:
Overlooked that one, grep should be fixed as well now.

Reply:
Still not working:
> grep: repetition-operator operand invalid

Reply:
Rewrite in deno should fix compatibility